File: library/src/main/java/com/amar/library/provider/ScreenInfoProvider.java
package com.amar.library.provider;

import android.content.Context;
import android.graphics.Point;
import android.util.DisplayMetrics;

import com.amar.library.provider.interfaces.IScreenInfoProvider;

import javax.annotation.ParametersAreNonnullByDefault;

/**
 * Created by Amar Jain on 17/03/17.
 */

@ParametersAreNonnullByDefault
public class ScreenInfoProvider implements IScreenInfoProvider{

    private final Context mContext;

    public ScreenInfoProvider(Context context) {
        mContext = context;
    }

    @Override
    public int getScreenHeight() {
        return getDeviceDimension().y;
    }

    @Override
    public int getScreenWidth() {
        return getDeviceDimension().x;
    }

    Point getDeviceDimension() {
        Point lPoint = new Point();
        DisplayMetrics metrics = mContext.getResources().getDisplayMetrics();
        lPoint.x = metrics.widthPixels;
        lPoint.y = metrics.heightPixels;
        return lPoint;
    }
}

File: library/src/main/java/com/amar/library/provider/ScreenInfoProvider.kt
package com.amar.library.provider

import android.content.Context
import android.graphics.Point
import com.amar.library.provider.interfaces.IScreenInfoProvider

internal class ScreenInfoProvider(private val mContext: Context) : IScreenInfoProvider {
    private val deviceDimension: Point
        get() {
            val lPoint = Point()
            val metrics = mContext.resources.displayMetrics
            lPoint.x = metrics.widthPixels
            lPoint.y = metrics.heightPixels
            return lPoint
        }
    override val screenHeight: Int
        get() = deviceDimension.y
    override val screenWidth: Int
        get() = deviceDimension.x

    override val navigationBarHeight: Int
        get() {
            val resourceId: Int =
                mContext.resources.getIdentifier("navigation_bar_height", "dimen", "android")
            return if (resourceId > 0) {
                mContext.resources.getDimensionPixelSize(resourceId)
            } else 0
        }
}